### ISSUE TITLE
MediaProjection on mekotronics TV box in portrait mode distorted

### ISSUE THREAD

**Issue by Moomoozer** [bk138/droidVNC-NG#249]
**Describe the bug**
Remote screen gets distorted when enabling fast mode after prompt.

**To Reproduce**
Installed DroidVnc as normal on an Android mekotronics TV box in portrait mode
used Avnc / RealVnc client apps for android

https://github.com/user-attachments/assets/ba637d5e-4c32-4cae-9241-19d37b755d31

**Expected Behavior**

When connecting the droidVNC, a prompt recommends fast mode which screws of the client view

**Logs/Backtraces**
N/A

**Environment**
 - droidVNC-NG version: 2.7.3
 - Android version: "Android TV OS 12"
 - Client-side OS and version: 12
 - VNC client and version: AVNC 2.6.3 Android / RealVnc 4.9.2 Android

**Additional context**
In addition to fixing the problem, please allow users to disable the nag screen with a simple switch option in the ui.

**Comment by bk138** [comment]
Thanks for reporting! Few questions:

* Does fast mode work when you start with it, i.e. give the screen share permission right at the start?
* Does the output of https://gitlab.com/vijai/screenrecorder look scrambled as well (uses the same screen recording API)?

**Comment by Moomoozer** [comment]
Not sure what you mean, fast mode has never worked without distortion on this particular device. The screenrecorder app worked fine full screen. I usually use XRecorder and it did exhibit the exact same distortion issue until I changed the setting from "Normal mode"  to "Performance mode", which is properly the same as giving users the option of disabling what you refer to as fast mode.

**Comment by bk138** [comment]
We'd need an open source screen recorder that works for you so I can look into their source code and apply the change to droidVNC-NG. I only know https://gitlab.com/vijai/screenrecorder so far, this worked for you?

**Comment by Moomoozer** [comment]
Yes, the screenrecorder app you referred to was tested and working out of the box except for the audio.

**Comment by bk138** [comment]
@Moomoozer Thanks for the swift reply, I'll have a look. 

It's a bit hard to see from your video due to the black background, would "landscape screen recording is set into portrait mode" describe it correctly?

Did you test screenrecorder in portrait mode?

**Comment by Moomoozer** [comment]
It was tested in portrait mode set to 90 degrees in Android system display settings

**Comment by bk138** [comment]
Note to self: https://gitlab.com/vijai/screenrecorder/-/blob/master/app/src/main/java/com/orpheusdroid/screenrecorder/services/RecorderService.java?ref_type=heads#L694 is how they do it.
